portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -74,28 +83,51 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       commentEntity.setProperty("name", name);
       commentEntity.setProperty("comment", comment);
       commentEntity.setProperty("timestamp", System.currentTimeMillis());
+      commentEntity.setProperty("length", comment.length());
 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       datastore.put(commentEntity);
 
-      // pause the thread to ensure data is in DB before page refresh
-      // could not find equivalent of async call to datastore (tried transactions
-      // but couldn't get them to work)
-      try {
-        Thread.sleep(10);
-      } catch (InterruptedException e) {
-        e.printStackTrace();
-      }
-
       response.sendRedirect("/index.html#Comments");
   }
 
   /**
-   * Converts the hardcoded ArrayList to JSON using Gson Java library
+   * Converts the List of Comment objects to JSON using Gson Java library
    */
   private String convertToJson(List<Comment> arr) {
     Gson gson = new Gson();
     String json = gson.toJson(arr);
     return json;
   }
+
+/**
+ * Prepares a query with the user-inputed sorting order
+ */
+  private Query prepareQuery(String sortingOrder) {
+      // set to the default case
```

Comment:
nit: consistent comments
most style guides recommend starting with a capital and ending with a period, always. It doesn't seem to be explicitly called out in the java style guide, so i'm less concerned about following that convention. However, let's try to be consistent: seems like most of this file starts comments with a capital and no period.
```suggestion
      // Set to the default case
```